Bottom: 0e538e52aa48e72636f3652f06f226bee3a540b6
Top:    7011a3a604819abdb1b827b72dd423bfb5410d23
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-29 16:54:31 +0100

hid/gtk: Hook up debug drawing APIs


---

diff --git a/src/hid/gtk/gtkhid-gdk.c b/src/hid/gtk/gtkhid-gdk.c
index a47803e..45f1e96 100644
--- a/src/hid/gtk/gtkhid-gdk.c
+++ b/src/hid/gtk/gtkhid-gdk.c
@@ -1299,3 +1299,26 @@ ghid_render_pixmap (int cx, int cy, double zoom, int width, int height, int dept
 
   return pixmap;
 }
+
+HID *
+ghid_request_debug_draw (void)
+{
+  /* No special setup requirements, drawing goes into
+   * the backing pixmap. */
+  return &ghid_hid;
+}
+
+void
+ghid_flush_debug_draw (void)
+{
+  ghid_screen_update ();
+  gdk_flush ();
+}
+
+void
+ghid_finish_debug_draw (void)
+{
+  ghid_flush_debug_draw ();
+  /* No special tear down requirements
+   */
+}
diff --git a/src/hid/gtk/gtkhid-main.c b/src/hid/gtk/gtkhid-main.c
index f5e2717..ac961db 100644
--- a/src/hid/gtk/gtkhid-main.c
+++ b/src/hid/gtk/gtkhid-main.c
@@ -2157,6 +2157,10 @@ hid_gtk_init ()
   ghid_hid.drc_gui                  = &ghid_drc_gui,
   ghid_hid.edit_attributes          = ghid_attributes;
 
+  ghid_hid.request_debug_draw       = ghid_request_debug_draw;
+  ghid_hid.flush_debug_draw         = ghid_flush_debug_draw;
+  ghid_hid.finish_debug_draw        = ghid_finish_debug_draw;
+
   hid_register_hid (&ghid_hid);
 #include "gtk_lists.h"
 }
diff --git a/src/hid/gtk/gui.h b/src/hid/gtk/gui.h
index 6cdeff2..d52d088 100644
--- a/src/hid/gtk/gui.h
+++ b/src/hid/gtk/gui.h
@@ -499,6 +499,9 @@ gboolean ghid_drawing_area_expose_cb (GtkWidget *, GdkEventExpose *,
 gboolean ghid_pinout_preview_expose (GtkWidget * widget, GdkEventExpose * ev);
 GdkPixmap *ghid_render_pixmap (int cx, int cy, double zoom,
                                int width, int height, int depth);
+HID *ghid_request_debug_draw (void);
+void ghid_flush_debug_draw (void);
+void ghid_finish_debug_draw (void);
 
 /* gtkhid-main.c */
 void ghid_pan_fixup (void);
